### PR TITLE
using docker

### PR DESCRIPTION
**Pull request by spencerdcarlson**

### CODE REVIEW

File: elastic.go [hunk 1]
```diff
@@ -5,14 +5,15 @@ import (
...
+	"strconv"
 	"time"
 
 	elastic "gopkg.in/olivere/elastic.v5"
 )
 
 // SetupElastic sets up elastic conn
 func SetupElastic() {
-	client, err := elastic.NewClient(elastic.SetURL(elasticURL()))
+	client, err := elastic.NewClient(elastic.SetURL(elasticURL()), elastic.SetSniff(sniff()))
```
Comment (spencerdcarlson) [inline elastic.go]: This was the issue with hitting elastic search in docker here is the [explanation](https://github.com/olivere/elastic/wiki/Sniffing)

**Comment by spencerdcarlson** [comment]
The test passes, but it still is not posting to elasticsearch. When I hit http://localhost:9200/pg-2020-03-28/_search in the browser, I can see all the other test messages except for the elixir.json payload

**Comment by spencerdcarlson** [comment]
All of the files under vendor are deleted because I switched over to use [go.mod](https://github.com/golang/go/wiki/Modules) for dependency management. `go build` will auto pull down the correct dependencies.

**Comment by spencerdcarlson** [comment]
weird ... i can see the this data getting posted to es, and the following stats. but no data in es
```bash
data: {"@timestamp":"2020-03-28T08:18:35.816567Z","application_name":"animal_farm","avg_duration_ms":17.646,"code_source":null,"command_tag":"EXECUTE","database_name":"winston","day_of_week":"Monday","day_of_week_int":1,"detail":[{"action":"open","anonymous_user":null,"appName":"Web App","context":"messenger","event":"messenger.conversation","feature":"conversation","meta":{"afSyncUID":"b5388e64-03d7-4f00-bbce-c5ced48e17d8"},"published_at":"2020-03-28T08:18:35.781384Z","session":{"ip":"","libraryName":"@podiumhq/podium-analytica","libraryVersion":"0.1.7","pagePath":null,"pageReferrer":"","pageSearch":null,"pageTitle":"(4) Podium | Inbox","pageUrl":"","userAgent":"insomnia/7.1.1"},"user_id":null,"user_uid":null},"2020-03-28T08:18:35Z","2020-03-28T08:18:35","2020-03-28T08:18:35"],"error_severity":"LOG","max_duration_ms":17.646,"metadata":{},"min_duration_ms":17.646,"prepared":"N/A","prepared_step":"execute","query":"INSERT INTO \"raw\".\"raw_events\" (\"data\",\"published_at\",\"inserted_at\",\"updated_at\") VALUES ($1,$2,$3,$4) RETURNING \"id\"","redis_key":"postgres_test","remote_host":"localhost","total_count":2,"total_duration_ms":35.292,"unique_sha":"e58b640726ae2acf3e5f6f08edfe974ffb602990","unique_string":"insert into \"raw\".\"raw_events\" (\"data\",\"published_at\",\"inserted_at\",\"updated_at\") values ($1,$2,$3,$4) returning \"id\"","user_name":"postgres"}
stats {Flushed:1 Committed:0 Indexed:0 Created:0 Updated:0 Deleted:0 Succeeded:0 Failed:0 Workers:[0xc00019c900]}
worker: &{Queued:0 LastDuration:0s}
```

**Comment by spencerdcarlson** [comment]
I found the issue. I'm not sure why, but it does not like a list of different types.

this payload won't post to es:

```json
"user_name": "new_samplepayload",
...
"detail": [
    {
      "action": "open",
      "meta": {
        "afSyncUID": "b5388e64-03d7-4f00-bbce-c5ced48e17d8"
      },
      "published_at": "2020-03-28T08:18:35.781384Z",
      "session": {
        "libraryName": "@podiumhq/podium-analytica",
        "libraryVersion": "0.1.7"
      }
    },
    "2020-03-28T08:18:35Z",
    "2020-03-28T08:18:35",
    "2020-03-28T08:18:35"
  ],
...
```
its a list with a complex map as the first item, and 3 strings (the 3 strings are breaking it)

this payload is valid to post to es:
```json
"user_name": "new_samplepayload",
...
"detail": [
    {
      "action": "open",
      "meta": {
        "afSyncUID": "b5388e64-03d7-4f00-bbce-c5ced48e17d8"
      },
      "published_at": "2020-03-28T08:18:35.781384Z",
      "session": {
        "libraryName": "@podiumhq/podium-analytica",
        "libraryVersion": "0.1.7"
      }
    },
    {"2": "2020-03-28T08:18:35Z"},
    {"3": "2020-03-28T08:18:35"},
    {"4": "2020-03-28T08:18:35"}
  ],
...
```

convert the 3 strings to maps ... not sure why this works, but it does

File: elastic.go (hunk 1 above)
Comment (brianbroderick) [reply elastic.go]: I wonder if this will cause issues in prod?

Comment (spencerdcarlson) [reply elastic.go]: it defaults to `true` which is what it was before

Comment (spencerdcarlson) [reply elastic.go]: >SetSniff(bool) allows you to specify whether Elastic should sniff the cluster periodically (default is true).

[Configuration](https://github.com/olivere/elastic/wiki/Configuration)

Comment (brianbroderick) [reply elastic.go]: Ah got it

**Pull request merged**